pkg/observer/controller.go

```diff
@@ -334,12 +381,31 @@ func (c *controller) Run() {
 		}
 	}
 
-	// wait for the desired amount to allow any in progress changes to batch up
+    // Filter to only the lowest priority nodegroups
+    lowestPriorityBatch := c.selectLowestPriorityNodeGroups(changedNodeGroups)
+    if len(lowestPriorityBatch) == 0 {
```

Comment:
nit: doesn't this imply that there are no nodegroups to apply at all? Otherwise you would always get at least 1 nodegroup in the lowest priority batch